### PR TITLE
Milestone 10: Add authorization

### PR DESCRIPTION
**Pull request by zdnahom**
## Milestone 10: Add authorization
In this milestone me and @gaks1 : 
- Added the cancancan gem to add authorization to our project
- We added a role column to the users table 
- Added a delete post feature where authorized users(if the user have an admin role or it is theirs) can delete their post
- Added a delete a comment feature where authorized users(if the user have an admin role or it is theirs) can delete their comments
- Added related buttons for the delete features where only authorized users can view

### CODE REVIEW

File: app/models/user.rb [hunk 1]
```diff
@@ -13,4 +13,8 @@ class User < ApplicationRecord
   def recent_posts
     posts.order(created_at: :desc).limit(3)
   end
+
+  def admin?
+    role == 'admin'
+  end
```
Comment (Felix45) [inline app/models/user.rb:19]: - [ Kudos] Good job creating this method inside the `User` model, this method will make it easy to determine whether a user is an `admin` or not :+1:

**Review by Felix45 — APPROVED** [review]
# Approved :green_circle: 
Hi @zdnahom & @gaks1, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



## Highlights :heavy_check_mark: 
- [x] Application uses `cancancan` gem correctly :white_check_mark: 
- [x] Authorization rules work correctly :white_check_mark: 
- [x] Pull request has a descriptive title and summary :white_check_mark: 
- [x] Git flow is used correctly :white_check_mark: 
- [x] Repository has a professional readme file :white_check_mark:    

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/models/user.rb (hunk 1 above)
Comment (zdnahom) [reply app/models/user.rb:19]: Thanks for the feedback @Felix45, and also thanks for noticing that 😉

**Comment by zdnahom** [comment]
Thanks for the review @Felix45, your feedback is very appreciated 🙏

**Pull request merged**